### PR TITLE
feat: added pagination in products and nodes

### PR DESCRIPTION
**Pull request by psin32**
## Type

* ### Feature
  Implements a new feature

## Description

Adds pagination support for products and nodes in the PXM Catalogs endpoint

## Dependencies

*Other PRs or builds that this PR depends on.*

## Issues

*A list of issues closed by this PR.*

* Fixes [MT-14434]

[MT-14434]: https://elasticpath.atlassian.net/browse/MT-14434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/endpoints/catalogs.js [hunk 1]
```diff
@@ -29,8 +29,15 @@ class Nodes {
...
   GetNodeChildren({ nodeId, token = null }) {
+    const { limit, offset, includes, sort, filter } = this
     return this.request.send(
-      `catalogs/${this.endpoint}/${nodeId}/relationships/children`,
+      buildURL(`catalogs/${this.endpoint}/${nodeId}/relationships/children`, {
+        includes,
+        sort,
+        limit,
+        offset,
+        filter
```
Comment (samblacklock) [inline src/endpoints/catalogs.js:40]: @psin32 are `includes`, `sort`, `filter` available on these endpoints?

**Review by samblacklock — COMMENTED** [review]
thanks @psin32 i left one small comment

were you able to validate the changes locally? I don't really have a test store with enough nodes to properly test the pagination!

File: src/endpoints/catalogs.js (hunk 1 above)
Comment (psin32) [reply src/endpoints/catalogs.js]: @samblacklock I have removed `includes` and `filter` from this.

**Comment by samblacklock** [comment]
> thanks @psin32 i left one small comment
> 
> were you able to validate the changes locally? I don't really have a test store with enough nodes to properly test the pagination!

tested locally by @psin32 and confirmed working.

**Review by samblacklock — APPROVED** [review]

**Pull request merged**